libraries/Microsoft.Bot.Builder.Core.Extensions/CatchExceptionMiddleware.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class CatchExceptionMiddleware : IMiddleware
+    {
+        private readonly CatchExceptionHandler _handler;
+
+        public CatchExceptionMiddleware(CatchExceptionHandler handler)
+        {
+            _handler = handler;
+        }
+
+        public async Task OnProcessRequest(IBotContext context, MiddlewareSet.NextDelegate next)
+        {
+            await CatchError(context, next);
```

Comment:
While I love the clarity, I think the CatchError method code could just be right here.

Reply:
Removed the catch error method and move code directly into OnProcessRequest